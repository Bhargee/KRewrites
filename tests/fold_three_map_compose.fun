fold (fun a -> fun b -> a + b) map (fun c -> c * 2) map (fun d -> d + 1) map
(fun e -> e * 4)
