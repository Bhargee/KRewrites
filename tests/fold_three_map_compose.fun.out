Search results:

Solution 1:
<T>
    <k>
        (((((((fold) (fun a -> (fun b -> (a + b)))) ((map) (fun c -> (c * 
          2)))) (map)) (fun d -> (d + 1))) (map)) (fun e -> (e * 4)))
    </k>
    <store>
        .Map
    </store>
    <env>
        .Map
    </env>
</T>

Solution 2:
<T>
    <k>
        (((((fold) ((fun a -> (fun b -> (a + b))) -| (fun c -> (c * 2)))) 
          ((map) (fun d -> (d + 1)))) (map)) (fun e -> (e * 4)))
    </k>
    <store>
        .Map
    </store>
    <env>
        .Map
    </env>
</T>

Solution 3:
<T>
    <k>
        (((fold) (((fun a -> (fun b -> (a + b))) -| (fun c -> (c * 2))) -|
           (fun d -> (d + 1)))) ((map) (fun e -> (e * 4))))
    </k>
    <store>
        .Map
    </store>
    <env>
        .Map
    </env>
</T>

Solution 4:
<T>
    <k>
        ((fold) ((((fun a -> (fun b -> (a + b))) -| (fun c -> (c * 2))) -|
           (fun d -> (d + 1))) -| (fun e -> (e * 4))))
    </k>
    <store>
        .Map
    </store>
    <env>
        .Map
    </env>
</T>
